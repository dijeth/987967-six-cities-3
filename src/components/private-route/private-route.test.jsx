import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter} from 'react-router-dom';
import PrivateRoute from './private-route.jsx';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`<PrivateRoute />`, () => {

  it(`should render Component when require is true`, () => {
    const tree = Enzyme.mount(
        <BrowserRouter>
          <PrivateRoute
            path="/"
            to="/dublin"
            render={() => <div>london</div>}
            require={true}
          />
        </BrowserRouter>
    );

    expect(tree.getDOMNode()).toMatchSnapshot();
  });

  it(`should render null when require is false`, () => {
    const tree = Enzyme.mount(
        <BrowserRouter>
          <PrivateRoute
            path="/"
            to="/dublin"
            render={() => <div>london</div>}
            require={false}
          />
        </BrowserRouter>
    );

    expect(tree.getDOMNode()).toMatchSnapshot();
  });
});
