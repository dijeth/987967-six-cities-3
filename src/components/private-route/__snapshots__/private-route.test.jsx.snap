// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> should render Component when require is true 1`] = `
<div>
  london
</div>
`;

exports[`<PrivateRoute /> should render null when require is false 1`] = `null`;
